scripts/setup_dev_environment.py

```
@@ -0,0 +1,53 @@
+Installs dependencies and sets up pre-commit hooks.
+"""
+
+import subprocess
+import sys
+from pathlib import Path
+
+def run_command(cmd: List[str], description: str) -> bool:
+    """Run a command and return success status."""
+    print(f"Setting up {description}...")
+    try:
+        result = subprocess.run(cmd, cwd=Path(__file__).parent.parent, check=True)
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)